### PR TITLE
edited the autocomplete function

### LINKED ISSUE #2: auto-complete js

### PR DESCRIPTION
**Pull request by abdallah-alshawaf**

### CODE REVIEW

File: public/fetch.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
-const xhr = new XMLHttpRequest()
 
 const fetch = (url, cb)=>{
+    const xhr = new XMLHttpRequest();
+    let searchParam = searchInput.value;
```
Comment (AhmedSafi97) [inline public/fetch.js:4]: Here, you should seperate the fetching logic from any dom implementation, searchParam value should be generic(passed as an argument to the fetch function along with the url)
```js
fetch(`${url}?q=${searchParam}`, console.log)
```

As you can see, it should be sent along in the querystring not as a custom header

File: public/DOM.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+
+const body = document.body;
```
Comment (AhmedSafi97) [inline public/DOM.js]: what is this line for?

File: public/DOM.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+
+const body = document.body; 
+const suggestionsDiv = document.querySelector("#suggestions-section");
+const searchButton = document.querySelector("#search-btn");
+const searchInput = document.querySelector("#search");
+
+// function expressions need to be defined before called in the fetch call below. 
+const renderData = (data) => {
+    // first line is instead of innerHTML = '' to empty the div;
```
Comment (AhmedSafi97) [inline public/DOM.js:8]: redundant comments, I can tell by reading your code

File: public/DOM.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        p.textContent = `${element}`;
+        
+        div.appendChild(p);
+        suggestionsDiv.appendChild(div);
+    })
+}
+
+searchButton.addEventListener('click', () => {
+    fetch('/suggestions', renderData);
+})
```
Comment (AhmedSafi97) [inline public/DOM.js]: leave an empty line at the end of the file

File: public/index.html [hunk 5]
```diff
@@ -38,6 +41,8 @@
 
     <!-- ********************* JS Links ********************* -->
     <script src="../public/main.js"></script>
+    <script src="../public/fetch.js"></script>
+    <script src="../public/DOM.js"></script>
```
Comment (AhmedSafi97) [inline public/index.html:45]: why would you add `../`? `/public/DOM.js` is enough and then you can handle this endpoint in your routers

File: public/DOM.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
+const fs = require('fs');
+const path = require('path');
```
Comment (AhmedSafi97) [inline public/DOM.js]: fs inside dom :eyes:?

File: public/index.html (hunk 5 above)
Comment (abdallah-alshawaf) [reply public/index.html:45]: I never thought of that, you're correct.

File: public/DOM.js (hunk 6 above)
Comment (abdallah-alshawaf) [reply public/DOM.js]: fixed!!

Comment (abdallah-alshawaf) [reply public/DOM.js]: thank you for your review ahmed, it really means a lot ♥

**Pull request merged**